### ISSUE TITLE
throttle: need a mechanism to define calls that have shared quotas

### ISSUE THREAD

**Issue by ericblade** [ericblade/mws-advanced#109]
example: GetReportList and GetReportListByNextToken share the same request quota.  Lib does not currently understand that.  Probably need to include a "throttle key" option in the throttle data that different calls can use to mark that they should all belong to the same Queue.

**Comment by pwrnrd** [comment]
> you've got two completely separate instances of your service running (as I do), they don't currently know about each other's quota usages at all

Exactly! Therefore, I was thinking about the information both of these instances should know, and that's the MWS response they retrieve. So I thought that we could use that information to scale the time between requests. However, this might also be bit overkill... In addition, the problem you mention regarding the quota headers not being send is tricky in case of multiple instances/applications. Therefore, I decided to rewrite my current implementation.

I think adding a throttle key as you describe makes sense for the simpler cases (shared quotas/single app with one or multiple instances).
